cmd/karmada-search/app/options/options.go

```diff
@@ -87,6 +87,13 @@ func (o *Options) Run(ctx context.Context) error {
 
 	server.GenericAPIServer.AddPostStartHookOrDie("start-karmada-search-informers", func(context genericapiserver.PostStartHookContext) error {
 		config.GenericConfig.SharedInformerFactory.Start(context.StopCh)
+
+		// start ResourceRegistry controller
+		ctl, err := search.NewController(restConfig, search.CachedResourceHandler())
+		if err != nil {
+			return err
+		}
+		ctl.Start(context.StopCh)
```

Comment:
How about adding another PostStartHookFunc to GenericAPIServer with its own name?